### ISSUE TITLE
[bug] Domains `getAll`

### ISSUE THREAD

**Issue by TisRyno** [matt-major/do-wrapper#40]
Hey,

Started using this library for a Git managed DNS and Firewall setup we are creating at work (we have probably 150 domains)

During my testing I have run into an issue specifically with the `domains.getAll()` functionality but I assume it affects many other `all` functions due to the problem existing within the `request-helper`

The issue is only apparently when running:
```js
doAPI.domains.getAll('', true)
        .then((response) => console.log(response))
        .catch((err) => console.log("Error:", err))
```

I receive an error with
```
TypeError: Cannot read property 'total' of undefined
at .../node_modules/do-wrapper/dist/request-helper.js:90:31
```

I took a look into the dist copy of my request-helper and played around until I found there to be multiple issues relating to how the Promises are managed for the multi-request setup.

Namely:
Request Helper:
```js
    private submitRequest(options: any, callback: any): void {
        const requestOptions = this.requestBuilder(options);

        got(this.apiUrl + options.actionPath, requestOptions)
            .then((response: any) => {
                callback(response.body);
            })
            .catch((error: any) => {
                callback(error);
            });
    }
```
The callback(response.body) does not match required callback in `getAllPages` which makes `total = body.meta.total;` attempt to fetch meta on undefined.

I also found that even if I fixed the callback the response was never run through a parse so it's attempting to fetch meta and total on a string which obviously doesn't work.

**Comment by TisRyno** [comment]
Hoping the below changes will help with understand what I mean and potentially can be used to fix the issue, I would add a PR for these changes but I haven't got the setup to test other use cases past domains and records.

`/src/request-helper.ts`
Lines 58-60
```js
            .then((response: any) => {
                callback(null, JSON.parse(response.body));
            })
```

Lines 90-115
```js
        this.submitRequest(options, (err: any, body: any) => {
            if (err) {
                callback(null, err);
            }

            total = body.meta.total;
            items = items.concat(body[key]);
            required = Math.ceil(total / (options.qs.per_page || 25));

            if (items.length >= total) {
                return callback(items);
            } else {
                this.getRemainingPages(options, 2, required, function (err: any, response: any, body: any) {
                    if (err) {
                        callback(null, err);
                    }

                    completed++;
                    items = items.concat(body[key]);

                    if (completed === required) {
                        callback(response);
                    }
                });
            }
        });
```

With the above changes, the callback to `submitRequest` drops support for 'response' being an outcome, fixes the callback from `submitRequest` to match the expected callback from `request`  which is (body, error)

Also makes sure body is parsed.

I might be entirely missing the idea or something as I rarely dabble in Promise stuff.

Thanks.

**Comment by matt-major** [comment]
@TisRyno yep this looks like a bit of an oversight, I'll look in to this and get it fixed.

**Comment by matt-major** [comment]
@TisRyno this should be fixed as of v4.0.0-alpha.4
